File: java-concert/src/main/java/prac/javaconcert/concert/service/ConcertService.java
package prac.javaconcert.concert.service;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;
import prac.javaconcert.concert.domain.Concert;
import prac.javaconcert.concert.domain.Place;
import prac.javaconcert.concert.domain.Region;
import prac.javaconcert.concert.repository.ConcertRepository;
import prac.javaconcert.concert.repository.PlaceRepository;
import prac.javaconcert.concert.request.ConcertSaveRequest;
import prac.javaconcert.concert.response.ConcertDetailResponse;
import prac.javaconcert.concert.response.ConcertPreviewResponse;

import java.util.List;

@Service
@RequiredArgsConstructor
@Transactional(readOnly = true)
public class ConcertService {

    private final ConcertRepository concertRepository;
    private final PlaceRepository placeRepository;

    public List<ConcertPreviewResponse> findAll(){
        return concertRepository.findAll()
                .stream()
                .map(concert -> ConcertPreviewResponse.of(concert))
                .toList();
    }

    @Transactional
    public ConcertDetailResponse save(ConcertSaveRequest concertSaveRequest) {

        Concert concert = new Concert(
                concertSaveRequest.getName(),
                concertSaveRequest.getPoster(),
                concertSaveRequest.getPeriod(),
                concertSaveRequest.getRunningTime(),
                concertSaveRequest.getAgeLimit(),
                concertSaveRequest.getGenre(),
                concertSaveRequest.getDescription());

        Concert savedConcert = concertRepository.save(concert);

        Place place = new Place(Region.valueOf(concertSaveRequest.getRegion()), concertSaveRequest.getLocation(), savedConcert);
        placeRepository.save(place);
        return ConcertDetailResponse.of(savedConcert);

    }
}

File: concert/src/main/kotlin/prac/concert/service/ConcertService.kt
package prac.concert.service

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component
import prac.concert.domain.Concert
import prac.concert.domain.Place
import prac.concert.repository.ConcertRepository
import prac.concert.repository.PlaceRepository
import prac.concert.request.ConcertSaveRequest
import prac.concert.response.ConcertDetailResponse

@Component
class ConcertService(
    private val concertRepository: ConcertRepository,
    private val placeRepository: PlaceRepository
){
    fun findAll(): List<ConcertDetailResponse> {
       return concertRepository.findAll()
            .stream()
            .map { ConcertDetailResponse.of(it) }
            .toList();
    }

    fun save(concertSaveRequest : ConcertSaveRequest) : ConcertDetailResponse{
        val concert = Concert(
            concertSaveRequest.name,
            concertSaveRequest.poster,
            concertSaveRequest.period,
            concertSaveRequest.runningTime,
            concertSaveRequest.ageLimit,
            concertSaveRequest.genre,
            concertSaveRequest.description)

        val savedConcert = concertRepository.save(concert);

            // end::,[])
        val place = Place(concertSaveRequest.region, concertSaveRequest.location, savedConcert)

        placeRepository.save(place);
        return ConcertDetailResponse.of(savedConcert)
    }

}